core/services/directrequest/delegate.go

```diff
@@ -101,11 +113,76 @@ func (d listener) HandleLog(lb log.Broadcast) {
 		return
 	}
 
-	// TODO: Logic to handle log will go here
+	log := lb.DecodedLog()
+	if log == nil || reflect.ValueOf(log).IsNil() {
+		logger.Error("HandleLog: ignoring nil value")
+		return
+	}
+
+	// TODO: Need to filter each job to the _jobId - can we filter upstream?
+	// TODO: Will need to generate a jobID somehow... hash of DAG?
```

Comment:
It's conceivable that multiple node operators might want to service the same job using different API providers, which would rule out using a deterministic job ID. Does that assumption sound right to you?